### PR TITLE
NMS-7188: Add an ActiveMQ Broker Detector and Monitor

### PR DESCRIPTION
**Pull request by roskens**
This PR adds support for using the ActiveMQ client libraries to monitor MQ brokers.

### All Contributors

* [X] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [X] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [x] Have you made a comment in that issue which points back to this PR?
* [X] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [X] If this is a new or updated feature, is there documentation for the new behavior?
* [X] If this is new code, are there unit and/or integration tests?

### Pull Request Process

One or more reviewers should be assigned to each PR.

If you know that a particular person is subject matter expert in the area your PR affects, feel free to assign one or more reviewers when you create this PR, otherwise reviewers will be assigned for you.

Once the reviewer(s) accept the PR and the branch passes continuous integration in Bamboo, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the Order of the Green Polo) you are welcome to merge the PR.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-7188
* Bamboo (Continuous Integration): https://bamboo.opennms.org/

### CODE REVIEW

File: opennms-doc/guide-admin/src/asciidoc/text/service-assurance/monitors/ActiveMQMonitor.adoc [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+.Monitor specific parameters for the ActiveMQMonitor
+[options="header, autowidth"]
+|===
+| Parameter        | Description                                                                                        | Required | Default value
+| `brokerURL`      | The ActiveMQ Broker URL to connect to.                                                             | required | `vm://localhost?create=false&broker.persistent=false`
+| `user`           | The user name used to login to the ActiveMQ broker.                                                | optional | `-`
+| `password`       | The password used to authenticate the user on the ActiveMQ broker.                                 | optional | `-`
+| `use-nodelabel`  | A boolean to enable using the nodelabel when connecting to the ActiveMQ broker.                    | optional | `false`
+| `create-session` | A boolean to enable creating a JMS Session when connecting to the ActiveMQ broker.                 | optional | `false`
+| `clientID`       | The client ID to use when connecting to the ActiveMQ broker.                                       | optional | `-`
```
Comment (j-white) [inline opennms-doc/guide-admin/src/asciidoc/text/service-assurance/monitors/ActiveMQMonitor.adoc]: Nit: Some parameter names use camel case i.e. `brokerURL`, whereas others use hyphens for separators i.e. `create-session`.

**Comment by mvrueden** [comment]
@roskens please adjust the monitor`s parameter names to either use camel case or hyphens in order to get this merged. Thank you. 

Added `triage` label to indicate that we are waiting for @roskens to address this.

File: opennms-doc/guide-admin/src/asciidoc/text/service-assurance/monitors/ActiveMQMonitor.adoc (hunk 1 above)
Comment (roskens) [reply opennms-doc/guide-admin/src/asciidoc/text/service-assurance/monitors/ActiveMQMonitor.adoc]: i was mixing parameters between those used specifically by the broker class object (brokerURL, clientID) and those by the monitor/detector. Switching to using only hyphens.

**Comment by pbrane** [comment]
> @roskens please adjust the monitor`s parameter names to either use camel case or hyphens in order to get this merged. Thank you.
> 
> Added `triage` label to indicate that we are waiting for @roskens to address this.

@roskens thanks for using hyphens!

**Pull request merged**